src/main/java/frc/robot/RobotContainer.java

```diff
@@ -107,4 +107,16 @@ private void configureButtonBindings() {
     public Command getAutonomousCommand() {
         return autonChooser.getSelected();
     }
+
+    public void periodic(){
+        //check if button presses, switch state accordingly
+        if (mechController.getAButtonPressed() == true){
+            if (gripper.state == Value.kForward){
+                gripper.state = Value.kReverse;
+            }
+            else {
+                gripper.state = Value.kForward;
+            }
```

Comment:
try using commands instead of making a periodic: see https://github.com/grt192/GRT2022/blob/b6262bdcf0f6f2c0bc4704c04ddbf55af00cbb7f/src/main/java/frc/robot/RobotContainer.java#L169 for an example. i think you could make a toggleClamp in the subsystem and make an instant command with gripper::toggleClamp?